src/boost.mk

```diff
@@ -60,29 +65,48 @@ define $(PKG)_BUILD
         --includedir='$(PREFIX)/$(TARGET)/include' \
         -sEXPAT_INCLUDE='$(PREFIX)/$(TARGET)/include' \
         -sEXPAT_LIBPATH='$(PREFIX)/$(TARGET)/lib' \
+        -sPTW32_INCLUDE='$(PREFIX)/$(TARGET)/include' \
+        -sPTW32_LIB='$(PREFIX)/$(TARGET)/lib' \
+        define="BOOST_THREAD_VERSION=4" \
+        define="BOOST_THREAD_DONT_USE_CHRONO" \
+        define="BOOST_THREAD_USES_DATETIME" \
+        define="BOOST_THREAD_DONT_PROVIDE_GENERIC_SHARED_MUTEX_ON_WIN" \
+        define="BOOST_THREAD_PROVIDES_NESTED_LOCKS" \
+        define="BOOST_THREAD_DONT_PROVIDE_ONCE_CXX11" \
         install
-
+    for lib in `ls "$(PREFIX)/$(TARGET)/lib"/libboost_*.a | tr "\n" " "`; \
+    do \
+    newlib=`echo \`basename $${lib}\` | $(AWK) '{gsub(/-mt-x[0-9]{2}/,"-mt"); gsub(/_pthread/,"")}1'`; \
+    echo ln -sf "$${lib}" "$(PREFIX)/$(TARGET)/lib/$${newlib}"; \
+    ln -sf "$${lib}" "$(PREFIX)/$(TARGET)/lib/$${newlib}"; \
+    done
```

Comment:
Don't rename the libs, it's failing here (x86_64-apple-darwin19.5.0):
```
for lib in `ls "/Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib"/libboost_*.a | tr "\n" " "`; do newlib=`echo \`basename ${lib}\` | /opt/local/bin/gawk '{gsub(/-mt-x[0-9]{2}/,"-mt"); gsub(/_pthread/,"")}1'`; echo ln -sf "${lib}" "/Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/${newlib}"; ln -sf "${lib}" "/Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/${newlib}"; done
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_atomic-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_atomic-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_chrono-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_chrono-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_container-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_container-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_context-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_context-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_contract-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_contract-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_coroutine-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_coroutine-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_date_time-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_date_time-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_exception-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_exception-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_filesystem-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_filesystem-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_graph-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_graph-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_iostreams-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_iostreams-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_locale-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_locale-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_log-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_log-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_log_setup-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_log_setup-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_math_c99-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_math_c99-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_math_c99f-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_math_c99f-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_math_c99l-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_math_c99l-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_math_tr1-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_math_tr1-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_math_tr1f-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_math_tr1f-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_math_tr1l-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_math_tr1l-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_prg_exec_monitor-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_prg_exec_monitor-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_program_options-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_program_options-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_random-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_random-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_regex-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_regex-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_serialization-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_serialization-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_stacktrace_basic-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_stacktrace_basic-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_stacktrace_noop-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_stacktrace_noop-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_system-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_system-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_test_exec_monitor-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_test_exec_monitor-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_thread_pthread-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_thread-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_timer-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_timer-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_type_erasure-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_type_erasure-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_unit_test_framework-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_unit_test_framework-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_wave-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_wave-mt-s-x32.a
ln -sf /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_wserialization-mt-s-x32.a /Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib/libboost_wserialization-mt-s-x32.a
# setup cmake toolchain
printf "set(Boost_THREADAPI "pthread")\\n set (Boost_USE_STATIC_LIBS ON)\\n set (Boost_USE_MULTITHREADED ON)\\n set (Boost_NO_BOOST_CMAKE OFF)\\n set (Boost_USE_STATIC_RUNTIME ON)\\n" > '/Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/share/cmake/mxe-conf.d/boost.cmake'
'i686-w64-mingw32.static-g++' -W -Wall -Werror -ansi -U__STRICT_ANSI__ -pedantic '/Users/tonyt/dev/mxe/src/boost-test.cpp' -o '/Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/bin/test-boost.exe' -std='c++11' -lboost_serialization-mt -lboost_thread-mt -lboost_system-mt -lboost_chrono-mt -lboost_context-mt -L'/Users/tonyt/dev/mxe/usr/i686-w64-mingw32.static/lib'
/Users/tonyt/dev/mxe/usr/bin/i686-w64-mingw32.static-ld: cannot find -lboost_serialization-mt
/Users/tonyt/dev/mxe/usr/bin/i686-w64-mingw32.static-ld: cannot find -lboost_thread-mt
/Users/tonyt/dev/mxe/usr/bin/i686-w64-mingw32.static-ld: cannot find -lboost_system-mt
/Users/tonyt/dev/mxe/usr/bin/i686-w64-mingw32.static-ld: cannot find -lboost_chrono-mt
/Users/tonyt/dev/mxe/usr/bin/i686-w64-mingw32.static-ld: cannot find -lboost_context-mt
collect2: error: ld returned 1 exit status
gmake[1]: *** [Makefile:846: build-only-boost_i686-w64-mingw32.static] Error 1
gmake[1]: Leaving directory '/Users/tonyt/dev/mxe'
```

Reply:
Goddamn it, this is a pain in the ass; I changed it like four times last night and eventually left the one that built fine here. Keep in mind, I'm not renaming, I'm adding symlinks.

But I see the issue, I think; the problem is the "s" static suffix; I think we do in fact need to rename them to remove that.

Reply:
I'd be happy to remove the manual test altogether and instruct users to use the provided cmake routines.

Reply:
That would probably be a more robust solution in the long-term.

Reply:
If only Boost had decent pkg-config support.